superset-frontend/src/SqlLab/components/RunQueryActionButton.tsx

```diff
@@ -17,53 +17,57 @@
  * under the License.
  */
 import React from 'react';
-import PropTypes from 'prop-types';
 import { t } from '@superset-ui/translation';
 
 import Button from '../../components/Button';
 
-const propTypes = {
-  allowAsync: PropTypes.bool.isRequired,
-  dbId: PropTypes.number,
-  queryState: PropTypes.string,
-  runQuery: PropTypes.func.isRequired,
-  selectedText: PropTypes.string,
-  stopQuery: PropTypes.func.isRequired,
-  sql: PropTypes.string.isRequired,
-};
-const defaultProps = {
-  allowAsync: false,
-  sql: '',
-};
+interface Props {
+  allowAsync: boolean;
+  dbId: number | null;
+  queryState: string;
+  runQuery: (c?: boolean) => void;
+  selectedText: string;
+  stopQuery: () => void;
+  sql: string;
+}
 
-export default function RunQueryActionButton(props) {
-  const runBtnText = props.selectedText
-    ? t('Run Selected Query')
-    : t('Run Query');
-  const btnStyle = props.selectedText ? 'warning' : 'primary';
-  const shouldShowStopBtn =
-    ['running', 'pending'].indexOf(props.queryState) > -1;
+const RunQueryActionButton = ({
+  allowAsync = false,
+  dbId = null,
+  queryState = '',
+  runQuery = () => {
```

Comment:
I think `() => undefined` is preferred for a NoOp function. Better yet, you should define a const at the top of the file for a no op function then use it as the defaults here like this:

`const NO_OP = () => undefined;`